moflask/flask.py

```diff
@@ -3,42 +3,48 @@
 import os
 
 from flask import Flask
-from werkzeug.utils import import_string
 
 from moflask.logging import init_logger
 
 
 class BaseApp(Flask):
     """Flask App."""
 
-    @staticmethod
-    def config_obj_from_env(env=None):
-        """Find a config object that matches the current environment."""
-        env = env or os.getenv("FLASK_ENV", "production").capitalize()
-        path = os.getenv("FLASK_CONFIG_OBJECT", "settings." + env + "Config")
-        return import_string(path)()
-
     def __init__(self, *args, **kwargs):
         """Create a new app object."""
-        config = kwargs.pop("config", None)
-        env = kwargs.pop("env", None)
+        config = kwargs.pop("config", {})
         super().__init__(*args, **kwargs)
-        # load app config
-        # 1. from an import path specfied in the FLASK_CONFIG_OBJECT
-        #    environmentfrom variable. Default: settings.DevelopmentConfig
-        # 2. A config object passed as keyword argument: config=Object
-        self.config.from_object(self.config_obj_from_env(env))
-        self.config.from_object(config)
-        init_logger(self)
+        self.sentry = None
+        self.load_config(config)
+        self.init_defaults()
 
         if not self.sanity_check():
             raise RuntimeError("Sanity checks failed. Aborting!")
 
+    def load_config(self, config=None):
+        """Load app config.
+
+        1. config file set in FLASK_SETTINGS environment variable (defaults to settings/base.py)
+        2. config object or dict passed as keyword argument
+        3. prefer "FLASK_" prefixed environment variables
```

Comment:
I'm wondering if maybe the app name (`self.config.from_prefixed_env(prefix=self.name.upper())`) would be a better prefix? It would make it really clear for which app the setting is intended and the same logic could be applied to python apps not using flask as well. On the other hand it makes it harder to re-use or copy-paste settings across apps and some app names might turn out to be a bit annoying as a prefix? :thinking:

Reply:
I had similar thoughts about LOGCRM_EVENT_BUS_ being a quite annoying prefix. It’s one of the things I aleady dislike about the roles.

Also it feels a bit weird to have different names for a variable that always affects the same piece of code (most of the configs are for extensions like flask-sqlalchemy, flask-jwt-extended, …

An argument for using a different prefix is that the environment variables that we already have (FLASK_APP) won’t be imported again.

How about prefixing with IMPACT_STACK_ or IST_? moflask could make that configurable by the sub-class.

Reply:
Hm, I don’t there is a huge difference between `IMPACT_STACK` and `LOGCRM_EVENT_BUS` anymore? :see_no_evil: And there are bunch of settings already starting with `IMPACT_STACK_`. `IST_IMPACT_STACK_API_URL` is also a bit weird?

I don’t find it weird to have different names for a variable that affects the same piece of code when the settings is actually different per app, e.g. `LOGCRM_SQLALCHEMY_DATABASE_URI`, `FORMTOOL_SQLALCHEMY_DATABASE_URI`. `IST_ SQLALCHEMY_DATABASE_URI` would make it sound like there is just one database for all apps.

We could use the app name and still make it configurable by the subclass?

Or we think about something even more generic, like `CONFIG_`, `APP_`,…?